packages/noya-renderer/src/components/DistanceLabelAndPath.tsx

```diff
@@ -0,0 +1,323 @@
+import {
+  Polyline,
+  Text,
+  useReactCanvasKit,
+  Rect as RCKRect,
+  useFontManager,
+  useColorFill,
+} from 'noya-react-canvaskit';
+import React, { useMemo } from 'react';
+import { createBounds, Point, Rect, distance } from 'noya-geometry';
+
+type DistanceMeasurementProps = {
+  distance: number;
+  bounds: Point;
+};
+
+function DistanceMeasurement({ distance, bounds }: DistanceMeasurementProps) {
+  const { CanvasKit } = useReactCanvasKit();
+  const fontManager = useFontManager();
+  const paragraph = useMemo(() => {
+    const paragraphStyle = new CanvasKit.ParagraphStyle({
+      textStyle: {
+        color: CanvasKit.parseColorString('rgb(255,255,225)'),
+        fontSize: 11,
+        fontFamilies: ['Roboto'],
+        letterSpacing: 0.2,
+      },
+    });
+
+    const builder = CanvasKit.ParagraphBuilder.Make(
+      paragraphStyle,
+      fontManager,
+    );
+    builder.addText(distance.toString());
+
+    const paragraph = builder.build();
+    paragraph.layout(10000);
+
+    return paragraph;
+  }, [CanvasKit, fontManager, distance]);
+
+  const labelRect = useMemo(
+    () =>
+      CanvasKit.XYWHRect(
+        bounds.x,
+        bounds.y,
+        paragraph.getMinIntrinsicWidth(),
+        paragraph.getHeight(),
+      ),
+    [CanvasKit, paragraph, bounds],
+  );
+
+  const bgRect = useMemo(
+    () =>
+      CanvasKit.XYWHRect(
+        bounds.x - 2,
+        bounds.y - 1,
+        paragraph.getMinIntrinsicWidth() + 5,
+        paragraph.getHeight() + 2,
+      ),
+    [CanvasKit, paragraph, bounds],
+  );
+
+  const backgroundFill = useColorFill('rgb(43, 92, 207)');
+
+  return (
+    <>
+      <RCKRect rect={bgRect} paint={backgroundFill} />
+      <Text rect={labelRect} paragraph={paragraph} />
+    </>
+  );
+}
+
+function MeasureAndExtensionPaths({
+  selectedLayer,
+  highlightedLayer,
+}: {
+  selectedLayer: Rect;
+  highlightedLayer: Rect;
+}) {
+  const highlightedBounds = createBounds(highlightedLayer);
+  const selectedBounds = createBounds(selectedLayer);
+
+  let implicitLines: Point[][] = [];
```

Comment:
Do we still want to use the naming we talked about when pairing? (I'm fine with either way)

> "Extension guides" - the lines extending edges of the highlighted layer
"Measurement guides" - the lines indicating a distance between two rects, that often show a text bubble with the distance value

Reply:
Oops! old habits die hard I guess, all names have been changed to `measureGuideLines` and `extensionGuideLines`